### PR TITLE
refactor navigator with unique routes rather than relaying on useLoca…

### PR DESCRIPTION
**Pull request by lawyu89**
…tion state

current routing is very bug and not robust because we relay on useLocation state. To resolve this, move paradigm to routes per page with ids of objects we need to fetch.

### CODE REVIEW

File: src/features/goals/models/Goal.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  description: string;
+  achievement_status: string;
+  start_on: string;
+  due_on: string;
+  cron_expression: string;
+  care_plan_id: string;
+  goal_template_id: string;
+  created_at: string;
+  upstringd_at: string;
+  cms_entry_id: null;
```
Comment (LUMacKenn) [inline src/features/goals/models/Goal.ts:15]:
```suggestion
  cms_entry_id: string;
```

File: src/features/goals/models/Goal.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  source_type: null;
+  note: null;
+  provider_response: Record<string, string>;
+}
+
+export interface Target {
+  id: string;
+  name: string;
+  description: string;
+  value: null;
```
Comment (LUMacKenn) [inline src/features/goals/models/Goal.ts:43]:
```suggestion
  observed_value: string;
  created_at: string;
  upstringd_at: string;
  observed_string: string;
  target_id: string;
  goal_id: string;
  observation_type_id: string;
  source_id: string;
  source_type: string;
  note: string;
  provider_response: Record<string, string>;
}

export interface Target {
  id: string;
  name: string;
  description: string;
  value: string;
```

File: src/features/goals/pages/Goal.tsx [hunk 3]
```diff
@@ -82,29 +80,31 @@ export default function Goal() {
         },
       },
     });
-    navigate("/goal", {
-      state: { goal: { ...goal, achievement_status: new_achievement_status } },
-    });
+    refetch();
```
Comment (LUMacKenn) [inline src/features/goals/pages/Goal.tsx:83]: could make this a mutation which invalidates that goal ID, no? Tbh I'm not sure if this is just a better pattern anyway

File: src/features/goals/pages/Goal.tsx [hunk 4]
```diff
@@ -82,29 +80,31 @@ export default function Goal() {
...
   };
 
-  if (isLoading) {
+  if (isLoading || goalLoading) {
     return <Skeleton variant="rectangular" animation="wave" height={280} />;
   }
 
+  if (goalError ?? !goal) {
```
Comment (LUMacKenn) [inline src/features/goals/pages/Goal.tsx:90]: Is "??" appropriate here? It totally might be just never seen it in this context for a boolean condition lol

File: src/features/tasks/pages/Task.tsx [hunk 5]
```diff
@@ -42,22 +38,18 @@ const Task = () => {
...
 
   if (taskError || taskWithCRMError) {
-    return <Navigate to="/" />;
+    return <Navigate to={`/home/${carePlanId}`} />;
   }
 
   return (
     <>
-      {taskWithCRM.imageUrl ? (
+      {task.imageUrl ? (
```
Comment (LUMacKenn) [inline src/features/tasks/pages/Task.tsx:46]: This is good as long as we keep the CRM loading condition

File: src/features/tasks/pages/Task.tsx [hunk 6]
```diff
@@ -42,22 +38,18 @@ const Task = () => {
...
         <Card sx={{ position: "relative" }}>
-          <ActionButton
-            type={"back"}
-            route={-1}
-            sx={{ position: "absolute", zIndex: 100 }}
-          />
-          <HeaderImage data={taskWithCRM} />
+          <ActionButton type={"back"} route={-1} sx={{ position: "absolute", zIndex: 100 }} />
```
Comment (LUMacKenn) [inline src/features/tasks/pages/Task.tsx:48]: Should we define this route as well?

**Review by LUMacKenn — APPROVED** [review]
Just some minor fixes but I'll approve so you can merge when you're ready!

File: src/features/goals/pages/Goal.tsx (hunk 3 above)
Comment (lawyu89) [reply src/features/goals/pages/Goal.tsx:83]: probably, i just did a quick fetch here for now.

File: src/features/goals/pages/Goal.tsx (hunk 4 above)
Comment (lawyu89) [reply src/features/goals/pages/Goal.tsx:90]: o good catch! this was test code, needed to refactor 😅

**Pull request merged**